virtual/types/types.go

```diff
@@ -45,9 +45,21 @@ type ActorReferenceVirtual interface {
 type ActorReferencePhysical interface {
 	// ServerID is the ID of the physical server that this reference targets.
 	ServerID() string
-	// The address of the referenced actor.
-	Address() string
 	// ServerVersion is incremented every time a server's heartbeat expires and resumes,
 	// guaranteeing the server's ability to identify periods of inactivity/death for correctness purposes.
 	ServerVersion() int64
+
+	// The state of the physical server that this reference targets.
+	// Contains information that is sent in the heartbeat.
+	ServerState() ServerState
+}
+
+
+type ServerState interface {
```

Comment:
Then we can get rid of all the `serializableServerState` stuff

Reply:
I understand your comment. I spent some time thinking about it too. I chose to use interfaces for two reasons:

1. The existing types were already interfaces, rather than structs.
2. The implementation of the state may vary depending on the registry or server. For example, the `ActorReferencePhysical` could be in different formats like protobuf or JSON, depending on the specific registry implementation. Furthermore, the underlying format may evolve into different versions over time. By using interfaces, we can abstract the versioning concerns and handle them through the encoder/decoder, without worrying about the specific implementation details.

However, I value your broader understanding of the project and would appreciate your insights on whether the second point aligns with the overall vision for NOLA.